Purchases/Public/RCPurchases.m

```diff
@@ -430,7 +427,11 @@ - (void)createAlias:(NSString *)alias completionBlock:(nullable RCReceivePurchas
             if (error == nil) {
                 [self updateAllCachesWithCompletionBlock:completion];
             } else {
-                CALL_IF_SET_ON_MAIN_THREAD(completion, nil, error);
+                if (completion) {
+                    [self.operationDispatcher dispatchOnMainThread: ^{
+                        completion(nil, error);
+                    }];
+                }
```

Comment:
the downside is that we now have 4 lines for each of these, instead of single-liners. 
the upside is the compiler checks for completion. 

Also, I'm being super conservative here, but a few of these `completions` might already be non-nil depending on the declaration in the header.

Reply:
So you don't think one private function that dispatchesOnMainThread if there's a completion would be worth it? We would also get the compiler checks right? Anyhow I think it looks good like this and it's petty clear, with the private function, we would only save the `if (completion)` check.

Reply:
my brain may be fried by now, but I couldn't come up with a nice way of doing that without going back to a macro and variadic arguements...
The issue is that we don't know the number or type of arguments that `completion` might have, if it's non-nil. 
but if `completion` is nil and you try to call it, you get an exception. 

So the method needs to receive a `nullable completion`, as well as its arguments, so it can do something like: 
```objective-c
if (completion) { 
    completion(arg1, arg2, arg3);
}
```

happy to chat over slack if this isn't clear. And please let me know if you think of a workaround, I'd love to be able to save that if check.

Reply:
note that each individual method can do this easily because it already knows the signature of `completion`

Reply:
@vegaro thoughts?

Reply:
you're totally right, I just tried to do it locally and understood what you mean. I guess that's why it was a macro 😆